frontend/src/ui/D3Visualizations/Treemap.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useLayoutEffect, useRef } from 'react';
+import * as d3 from 'd3';
+import { TreeHierarchy } from '../../domain/hierarchy';
+
+export type TreemapProps = {
+  height: number;
+  width: number;
+  data: TreeHierarchy;
+};
+
+export default function Treemap({ height, width, data }: TreemapProps) {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  useLayoutEffect(() => {
+    renderTreemap();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Doing this will prevent the viz from updating if any props change. you can wrap renderTreemap in useCallback so that it only re-renders when props change.